Products/Zing/datamaps.py

```diff
@@ -153,10 +153,14 @@ def _generate_facts(self, facts_per_device, zing_tx_state):
                 # organizers and impact relationships facts for the component
                 comp_uuid = f.metadata.get(ZFact.FactKeys.CONTEXT_UUID_KEY, "")
                 if comp_uuid:
+                    comp_groups = []
+                    for component in device.getDeviceComponents():
+                        if component.getUUID() == comp_uuid:
+                            comp_groups = component.getComponentGroupNames()
```

Comment:
After retrieving the component group names, there should be a `break` here to stop the `for` loop.  There's no point to continue looping, is there?